Realm/RLMSyncUser.mm

```diff
@@ -300,8 +313,14 @@ - (void)revokePermission:(RLMSyncPermissionValue *)permission callback:(RLMPermi
 
 #pragma mark - Private API
 
-- (void)_unregisterRefreshHandleForURLPath:(NSString *)path {
-    [self.refreshHandles removeObjectForKey:path];
+- (CocoaSyncUserContext *)_context {
+    if (!_user) {
+        return nullptr;
+    }
+    if (!_user->context) {
+        _user->context = std::make_unique<CocoaSyncUserContext>();
```

Comment:
This needs to be synchronized, I think. What if two different `RLMSyncUser`s try to do this at once?